trace_context/HTTP_HEADER_FORMAT.md

```diff
@@ -189,13 +189,25 @@ tracestate: rojo=00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01,congo=l
 
 ## Base mutations
 
-Library or platform receiving `traceparent` request header MUST send it to outgoing requests. It MAY mutate the value of this header before passing to outgoing requests. 
+Library or platform receiving `traceparent` request header MUST send it to outgoing requests. It MAY mutate the value of this header before passing to outgoing requests.
 
 If the value of `traceparent` field wasn't changed before propagation - `tracestate` MUST NOT be modified as well. Unmodified headers propagation is typically implemented in a pass-thru services like proxies. This behavior may also be implemented in a services which currently does not collect distributed tracing information.
 
 Here is the list of allowed mutations:
+
 1. **Update `span-id`**. The value of property `span-id` can be regenerated. This is the most typical mutation and should be considered a default.
 2. **Mark trace for sampling**. The value of `sampled` flag of `trace-options` may be set to `1` if it had value `0` before. `span-id` MUST be regenerated with the `sampled` flag update. This mutation typically happens to mark the importance of a current distributed trace collection.
 3. **Restarting trace**. All properties - `trace-id`, `span-id`, `trace-options` are regenerated. This mutation is used in the services defined as a front gate into secure network and eliminates a potential denial of service attack surface. 
 
-Libraries and platforms MUST NOT make any other mutations to the `traceparent` header.
\ No newline at end of file
+Libraries and platforms MUST NOT make any other mutations to the `traceparent` header.
+
+# Mutating the tracestate field
+
+Library or platform receiving `tracestate` request header MUST send it to outgoing requests. It MAY mutate the value of this header before passing to outgoing requests. The main concept of `tracestate` mutations is that order of unmodified key-value pairs SHOULD be preserved. Modified keys SHOULD be moved to the beginning of the list.
+
+Single request may be processed by multiple tracing systems in a single component. So any number of allowed mutations may happen in a single component. Here is the list of allowed mutations:
+
+1. **Update key value**. The value of any key can be updated. Modified key MUST be moved to the beginning of the list. This is the most common mutation.
+2. **Delete the key-value pair**. Any key-value pair MAY be deleted. It is highly discouraged to delete keys that wasn't generated by the same tracing system or platform. Deletion of unknown key-value pairs will break correlation in other system.
+3. **Make key-value pair first**. Tracing system MAY move any key-value pair to the first position. This mutation will indicate to the downstream which tracing system or platform sent this request.
```

Comment:
based on implementation I would suggest this:

1. Insert trace state entry <- you are starting a trace in beginning position
2. Update trace state entry < you are resuming a trace, by modifying your state, and moving it to position zero again
2. Delete trace state entry <- no idea why we document this? This is more like truncate

FYI: Please do not rush this to merge. I am pretty sure no-one is implementing this besides tyler and me so listen instead of rush language that conflicts with how things work

Reply:
> Delete trace state entry <- no idea why we document this? This is more like truncate

Mostly for [privacy](https://github.com/w3c/distributed-tracing/pull/110). And to enable potential truncation. I'm calling it explicitly so one will not be out of compliance by implementing the truncation or privacy sweep.

For other items - are you suggesting change of name for mutations? Or your feedback that you do not see the reason for unmodified mutation?